components/server/src/main/java/com/hotels/styx/server/netty/NettyServer.java

```diff
@@ -78,53 +73,34 @@
         this.host = nettyServerBuilder.host();
         this.channelGroup = requireNonNull(nettyServerBuilder.channelGroup());
         this.serverEventLoopFactory = requireNonNull(nettyServerBuilder.serverEventLoopFactory(), "serverEventLoopFactory cannot be null");
-        this.handlerFactory = requireNonNull(nettyServerBuilder.handlerFactory());
+        this.handler = requireNonNull(nettyServerBuilder.handler());
 
-        this.httpConnector = nettyServerBuilder.httpConnector();
-        this.httpsConnector = nettyServerBuilder.httpsConnector();
+        this.protocolConnector = nettyServerBuilder.protocolConnector();
 
-        this.httpServerSocketBinder = httpConnector.map(ServerSocketBinder::new).orElse(null);
-        this.httpsServerSocketBinder = httpsConnector.map(ServerSocketBinder::new).orElse(null);
-
-        this.startupActions = nettyServerBuilder.startupActions();
+        this.protocolSocketBinder = new ServerSocketBinder(protocolConnector);
     }
 
     @Override
     public InetSocketAddress httpAddress() {
-        if (httpServerSocketBinder != null) {
-            return new InetSocketAddress(host, httpServerSocketBinder.port());
-        } else {
-            return null;
-        }
-    }
-
-    @Override
-    public InetSocketAddress httpsAddress() {
-        if (httpsServerSocketBinder != null) {
-            return new InetSocketAddress(host, httpsServerSocketBinder.port());
-        } else {
-            return null;
-        }
+        return Optional.ofNullable(protocolSocketBinder)
+                .map(it -> {
+                    try {
+                        return new InetSocketAddress(host, it.port());
+                    } catch (IllegalArgumentException e) {
+                        return null;
```

Comment:
Can we at least log anything here?

Reply:
No.

This is not necessarily an erroneous situation. `NettyServer` is a stateful object. Returning `null` implies the server is shut (or not started). Whether or not this is acceptable depends on caller's use case. Logging a message therefore needs to take place at call site.